parts/linux/cloud-init/artifacts/cse_config.sh

```diff
@@ -91,6 +91,34 @@ ensureAuditD() {
   fi
 }
 
+{{- if ShouldConfigTransparentHugePage}}
+configureTransparentHugePage() {
+    THP_CONFIG_PATH="/etc/sysfs.conf"
+    THP_ENABLED={{GetTransparentHugePageEnabled}}
+    if [[ "${THP_ENABLED}" != "" ]]; then
+        echo "${THP_ENABLED}" > /sys/kernel/mm/transparent_hugepage/enabled
+        echo "kernel/mm/transparent_hugepage/enabled=${THP_ENABLED}" >> ${THP_CONFIG_PATH}
+    fi
+    THP_DEFRAG={{GetTransparentHugePageDefrag}}
+    if [[ "${THP_DEFRAG}" != "" ]]; then
+        echo "${THP_DEFRAG}" > /sys/kernel/mm/transparent_hugepage/defrag
+        echo "kernel/mm/transparent_hugepage/defrag=${THP_DEFRAG}" >> ${THP_CONFIG_PATH}
+    fi
+}
+{{- end}}
+
+{{- if ShouldConfigSwapFile}}
+configureSwapFile() {
+    SWAP_SIZE_KB=$(expr {{GetSwapFileSizeMB}} \* 1000)
+    DISK_FREE_KB=$(df /dev/sda1 | sed 1d | awk '{print $4}')
```

Comment:
just curious is /dev/sda1 guaranteed to exist and is it the only possible to store the swap file?

Reply:
also do we have any protection of the swap file size? it cannot be too large to make the vm not working?

Reply:
I think /dev/sda1 should be guaranteed since /dev/sda is the OS disk. I'm checking with Jorge to double confirm. 
We may create swap on other partitions but I think for now hard coded /dev/sda1 should be fine, as the first iteration.
We don't have extra protection other than free disk space check. Swap is by default off for k8s, also not recommended. I think customer should be aware of this when they try to enable it, but we should probably document it well.

Reply:
yes, documentation is important here.

Reply:
also let's test in both 1604 and 1804

Reply:
I've changed the script a little bit.
- partition has been changed to /dev/sdb1 per Jorge suggested. This is mounted at /mnt by default. sda and sdb are default disks Azure will provide https://docs.microsoft.com/en-us/azure/virtual-machines/linux/optimization#your-vm-temporary-drive
- retry at each step instead of func since they have dependencies. For example if fallocate fails we should just quit early.
- tested in both 1604 and 1804 nodes, worked as expected.